src/arrayMethodSort.js

```diff
@@ -4,8 +4,23 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(
+    compareFunction = (a, b) => String(a) > String(b)
+  ) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        const compareResult = compareFunction(this[j], this[j + 1]);
+
+        if (compareResult > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
+
+    return this;
```

Comment:
The method modifies the original array. It's better to return a new sorted array, leaving the original array untouched.

Reply:
https://mate.academy/chats/27190?selectedMessageId=1401459